hyperspy/models/model1D.py

```diff
@@ -289,49 +292,26 @@ def update_plot(self, *args, **kwargs):
             except:
                 self._disconnect_parameters2update_plot()
 
-    def suspend_update(self):
-        """Prevents plot from updating until resume_update() is called
+    @contextmanager
+    def suspend_update(self, update_on_resume=True):
+        """Prevents plot from updating until with clause completes.
 
         See Also
         --------
-        resume_update
         update_plot
         """
-        if self._suspend_update is False:
-            self._suspend_update = True
-            self._disconnect_parameters2update_plot()
-        else:
-            warnings.warn("Update already suspended, does nothing.")
-
-    def resume_update(self, update=True):
-        """Resumes plot update after suspension by suspend_update()
 
-        Parameters
-        ----------
-        update : bool, optional
-            If True, also updates plot after resuming (default).
+        all_comp_param_events = Events()
+        all_comp_param_events.children = lambda: [c.events for c in self]
+        old = self._suspend_update
+        self._suspend_update = True
+        with all_comp_param_events.suppress_hierarchy():
```

Comment:
This is actually very dangerous. You are suppressing all the events of all the components in the model. Currently only the functionality that you want to pause is connected by in the future we may have other events that should not be disconnected. Also, users may connect to the event a don't expect pausing the plotting update to suppress the `value_changed` event.